test/python/synthesis/test_synthesis.py

```diff
@@ -1270,6 +1270,15 @@ def test_use_dag(self, euler_bases, kak_gates, seed):
             requested_basis = set(oneq_gates + [kak_gate_name])
             self.assertTrue(decomposition_basis.issubset(requested_basis))
 
+    def test_non_std_gate(self):
+        """Test that the TwoQubitBasisDecomposer class can be correctly instantiated with a
+        non-standard gate.
+
+        Reproduce from: https://github.com/Qiskit/qiskit/issues/12998
+        """
+        # note that `CXGate(ctrl_state=0)` is not handled as a "standard" gate.
+        TwoQubitBasisDecomposer(CXGate(ctrl_state=0))(CXGate())
```

Comment:
The bug was only present when building the circuit. Can you update this to pass it a unitary matrix to synthesize and maybe assert the output circuit's operator is equivalent to the input and your pygate is correctly used.

Reply:
That's totally on me, I just copied the reproducing example without trying to make it readable. It should look better now :)